### PR TITLE
strategy implementation

### PR DESCRIPTION
**Pull request by Danielsio**

### CODE REVIEW

File: FacebookWinFormsApp/random strategy/PostRandomStrategy.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+namespace BasicFacebookFeatures.random_strategy
+{
+    internal class PostRandomStrategy : IRandomStrategy
+    {
+        private readonly PostViewer r_PostViewer;
+        private readonly TextBox r_TextBoxUploadDate;
+
+        public PostRandomStrategy(PostViewer i_PhotoViewer, TextBox i_TextBoxUploadDate)
+        {
+            this.r_PostViewer = i_PhotoViewer;
```
Comment (itayf9) [inline FacebookWinFormsApp/random strategy/PostRandomStrategy.cs]: Need to change the name of `i_PhotoViewer` to "i_PostViewer"

File: FacebookWinFormsApp/FormMain.cs [hunk 2]
```diff
@@ -694,17 +697,19 @@ private void fetchAlbumPictures(Album i_SelectedAlbum)
...
+            //    r_NostalgiaViewers[(int)eNostalgiaViewerIndex.PostViewer].SetVisibility(!v_ToBeVisible);
+            //    showNostalgiaPhoto();
+            // }
+            // else if (comboBoxMediaType.Text == k_ContentCategoryPosts)
+            // {
+            //    r_NostalgiaViewers[(int)eNostalgiaViewerIndex.PhotoViewer].SetVisibility(!v_ToBeVisible);
+            //    showNostalgiaPost();
+            // }
+            SetRandomStrategy();
+            m_RandomStrategy.ShowRandomContent(r_LoggedInUser);
```
Comment (itayf9) [inline FacebookWinFormsApp/FormMain.cs]: You need to turn off the visibility of the other viewer when you show some viewer. In this case you can see that in lines 703 and 708 we've turned the other viewer off, but you haven't added this in your refactor and that may cause the bug you've mentioned.
I suggest adding the following method `hideViewers()` and invoke it after line 711:

```c#
private void hideViewers()
{
         const bool v_ToBeVisible = true;

         if (comboBoxMediaType.Text == k_ContentCategoryPhotos)
        {
            r_NostalgiaViewers[(int)eNostalgiaViewerIndex.PostViewer].SetVisibility(!v_ToBeVisible);
        
        }
        else if (comboBoxMediaType.Text == k_ContentCategoryPosts)
        {
            r_NostalgiaViewers[(int)eNostalgiaViewerIndex.PhotoViewer].SetVisibility(!v_ToBeVisible);
        }
}

```

File: FacebookWinFormsApp/random strategy/PostRandomStrategy.cs (hunk 1 above)
Comment (Danielsio) [reply FacebookWinFormsApp/random strategy/PostRandomStrategy.cs]: i already fixed thoes issues but for some reason forgot to index them in the commit.. :)

File: FacebookWinFormsApp/FormMain.cs (hunk 2 above)
Comment (Danielsio) [reply FacebookWinFormsApp/FormMain.cs]: yep, that was it, it works now

**Comment by Danielsio** [comment]
let me know if you want to merge this one, i'm done working on it for now.

one notice is that i didnt use an object like DataFetcher we talkied about becasuse the whole Thread managment looked like hell of code abuse and i think we can address this issue later if still relevent

**Review by itayf9 — APPROVED** [review]
Looks good

**Pull request merged**